app/javascript/application.js

```diff
@@ -0,0 +1,14 @@
+// Entry point for the build script in your package.json
+import "@hotwired/turbo-rails"
+import "./controllers"
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+function App() {
```

Comment:
- [ ] I think we need to set up routing(from react-router) in this component that leads to the Greeting component hence you'll be required to create the missing `Greeting` component for this specific.
- [ ] I suggest you connect your app properly. The routes `/random_greeting` produces no method error on the console. Kindly make sure that you have an action, reducer, and a store for state management as requested in the requirements. The reducer should use the URL from Rails API. The two apps should connect together and work properly and random greetings should be displayed when the page is reloaded instead of the "Hello World" static text